app/assets/javascripts/discourse/app/routes/login.js

```diff
@@ -56,12 +56,16 @@ export default class extends DiscourseRoute {
       }
     }
 
-    // When Discourse Connect is enabled, redirect to the SSO endpoint
-    if (auth_immediately && enable_discourse_connect) {
-      const returnPath = cookie("destination_url")
-        ? getURL("/")
-        : encodeURIComponent(url);
-      window.location = getURL(`/session/sso?return_path=${returnPath}`);
+    // Handle DiscourseConnect first...
+    if (enable_discourse_connect) {
+      if (auth_immediately || !login_required) {
+        const returnPath = cookie("destination_url")
+          ? getURL("/")
+          : encodeURIComponent(url);
+        window.location = getURL(`/session/sso?return_path=${returnPath}`);
+      } else {
+        router.replaceWith("discovery.login-required");
+      }
```

Comment:
I think we can simplify this, regardless of `auth_immediately` or `login_required`, we should redirect to the SSO here on the `/login` or `/signup` routes for DiscourseConnect. When DiscourseConnect is enabled, it is the only auth mechanism, so there is no value in respecting `auth_immediately` in this route. 

On the "/" endpoint, it still decides whether a splash screen is shown (`auth_immediately=false`) or not (`auth_immediately=true`) for login required sites.